src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceMetadataBehavior.cs

```diff
@@ -162,6 +162,14 @@ private void SetExtensionProperties(ServiceDescription description, ServiceHostB
             }
         }
 
+        private static Uri GetFirstEndpointUriForScheme(string uriScheme, Uri configuredGetUrl, ServiceDescription description)
+            => configuredGetUrl
+                ?? description?.Endpoints?.FirstOrDefault(endpoint =>
+                       endpoint?.Address?.Uri != null
+                       && endpoint.Address.Uri.Scheme == uriScheme)
+                   ?.Address.Uri
```

Comment:
You are accepting that there might not be a matching endpoint and FirstOfDefault could return the default (which is null) by using `?.Address`, but if it's null, that will also evaluate to null, so `.Uri` will throw NullReferenceException.
```suggestion
                   ?.Address?.Uri
```